CondCore/PopCon/interface/OnlinePopCon.h

```diff
@@ -0,0 +1,113 @@
+#ifndef CONDCORE_POPCON_ONLINEPOPCON_H
+#define CONDCORE_POPCON_ONLINEPOPCON_H
+
+//
+// Authors:
+//  - Francesco Brivio (Milano-Bicocca)
+//  - Jan Chyczynski   (AGH University of Krakow)
+//
+
+#include "CondCore/DBOutputService/interface/OnlineDBOutputService.h"
+#include "CondCore/CondDB/interface/Time.h"
+#include "CondCore/PopCon/interface/Exception.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSetfwd.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include <algorithm>
+#include <functional>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace popcon {
+
+  //  Populator of the Condition DB.
+  //  Specific implementation for online lumi-based conditions (OnlineDBOutputService)
+
+  class OnlinePopCon {
+  public:
+    typedef cond::Time_t Time_t;
+
+    OnlinePopCon(const edm::ParameterSet& pset);
+
+    virtual ~OnlinePopCon();
+
+    template <typename Source>
+    void write(Source const& source);
+
+  private:
+    cond::persistency::Session initialize();
+    cond::persistency::Session preparePopCon();
+    void finalize();
+
+  private:
+    // DB service
+    edm::Service<cond::service::OnlineDBOutputService> m_dbService;
+
+    // PopCon infrastructure
+    cond::persistency::Session m_targetSession;
+    std::string m_targetConnectionString;
+    std::string m_authPath;
+    int m_authSys;
+    std::string m_recordName;
+    cond::TagInfo_t m_tagInfo;
+    cond::LogDBEntry_t m_logDBEntry;
+
+    // OnlinePopCon specific
+    int m_dbLoggerReturn_;  // DB logger return value
+    bool m_useLockRecors;   // whether to lock the records or not
+
+    // Version
+    static constexpr const char* const s_version = "5.0";
```

Comment:
Should the version be 5.0 starting from the original PopCon's version on rather 1.0?

Reply:
IIUC in the original PopCon the `s_version` was introduced 8 years ago directly with `"5.0"`, but I agree it's better to start with `"1.0"` for OnlinePopCon